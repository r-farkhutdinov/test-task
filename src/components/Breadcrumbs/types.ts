export type Breadcrumb = {
    id: string;
    title: string;
}

export interface BreadcrumbsProps {
    items?: Breadcrumb[];
}

export const breadcrumbsRootTestId = 'breadcrumbs-container';