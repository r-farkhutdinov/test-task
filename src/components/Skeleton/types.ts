export interface SkeletonProps {
    width?: number;
    isPaddingRight?: boolean;
    level?: number;
}
