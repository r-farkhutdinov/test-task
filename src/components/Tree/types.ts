export interface TreeProps {
    pages?: string[];
    isActive?: boolean;
    isActiveLevelTwo?: boolean;
}