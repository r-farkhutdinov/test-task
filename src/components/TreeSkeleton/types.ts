export interface TreeSkeletonProps {
    count?: number;
}