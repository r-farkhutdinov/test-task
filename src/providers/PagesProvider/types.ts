import { ReactNode } from 'react';
import { TableData } from '../../types';

export type PagesContextValue = {
    data?: TableData;
    activePage?: string;
    query?: string;
    loading?: boolean;
    error?: string;
    changeActivePage?: (activePage: string) => void;
    setQuery?: (query: string) => void;
    load?: (q?: string) => void;
};

export type Props = {
    children: ReactNode;
}