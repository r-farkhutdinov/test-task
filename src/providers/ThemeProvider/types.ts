import { ReactNode } from 'react';

export enum Theme {
    Dark = 'dark',
    Light = 'light'
}

export type ThemeContextValue = {
    theme: Theme;
    toggleTheme?: VoidFunction;
};

export type Props = {
    children: ReactNode;
}