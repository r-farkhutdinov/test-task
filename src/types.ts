export type Page = {
    id: string;
    title: string;
    level: number;
    parentId: string;
    url?: string;
    pages?: string[];
};

export type Entity = {
    pages: Record<string, Page>;
};

export type TableData = {
    entities: Entity;
    topLevelIds: string[]
};