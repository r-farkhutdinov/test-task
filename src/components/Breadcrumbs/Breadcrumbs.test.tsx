import { render, screen } from '@testing-library/react';
import { type Breadcrumb, breadcrumbsRootTestId } from './types';
import { Breadcrumbs } from '.';

const items: Breadcrumb[] = [
    {
        id: '1',
        title: 'Item 1'
    },
    {
        id: '2',
        title: 'Item 2'
    },
    {
        id: '3',
        title: 'Item 3'
    },
];

describe('Breadcrumbs component', () => {
    it('matches the snapshot with items', () => {
        const { asFragment } = render(<Breadcrumbs items={items} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches the snapshot without items', () => {
        const { asFragment } = render(<Breadcrumbs />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render breadcrumb items correctly', () => {
        render(<Breadcrumbs items={items.slice(0, 2)} />);

        expect(screen.getByTestId(`${breadcrumbsRootTestId}/item-1`)).toHaveTextContent(items[0].title);
        expect(screen.getByTestId(`${breadcrumbsRootTestId}/item-2`)).toHaveTextContent(items[1].title);
        expect(screen.getByTestId(breadcrumbsRootTestId)).toContainElement(screen.getByTestId(`${breadcrumbsRootTestId}/separator`));
    });

    it('should make the last breadcrumb active', () => {
        render(<Breadcrumbs items={items.slice(0, 2)} />);

        expect(screen.getByTestId(`${breadcrumbsRootTestId}/item-2`).classList.contains('active')).toBe(true);
    });

    it('should render empty message when no items present', () => {
        render(<Breadcrumbs />);
        expect(screen.getByTestId(`${breadcrumbsRootTestId}/empty`)).toHaveTextContent('Select two levels or more to see the breadcrumbs');
    });
});
