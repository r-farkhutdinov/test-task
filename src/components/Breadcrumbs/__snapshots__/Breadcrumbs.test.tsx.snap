// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component should match the snapshot with items 1`] = `
<DocumentFragment>
  <div
    class="root"
    data-testid="breadcrumbs-container"
  >
    <p
      class=""
      data-testid="breadcrumbs-container/item-1"
    >
      Item 1
    </p>
    <p
      data-testid="breadcrumbs-container/separator"
    >
      /
    </p>
    <p
      class=""
      data-testid="breadcrumbs-container/item-2"
    >
      Item 2
    </p>
    <p
      data-testid="breadcrumbs-container/separator"
    >
      /
    </p>
    <p
      class="active"
      data-testid="breadcrumbs-container/item-3"
    >
      Item 3
    </p>
  </div>
</DocumentFragment>
`;

exports[`Breadcrumbs component should match the snapshot without items 1`] = `
<DocumentFragment>
  <div
    class="root"
    data-testid="breadcrumbs-container"
  >
    <p
      data-testid="breadcrumbs-container/empty"
    >
      Select two levels or more to see the breadcrumbs
    </p>
  </div>
</DocumentFragment>
`;
